plugins/plugin-chart-echarts/src/Graph/transformProps.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  CategoricalColorNamespace,
+  ChartProps,
+  getMetricLabel,
+  DataRecord,
+} from '@superset-ui/core';
+import { EChartsOption, GraphSeriesOption } from 'echarts';
+import { GraphNodeItemOption } from 'echarts/types/src/chart/graph/GraphSeries';
+import { EchartsGraphFormData, DEFAULT_FORM_DATA as DEFAULT_GRAPH_FORM_DATA } from './types';
+import { GraphConstants, tooltipConfig, normalizationLimits } from './constants';
+import { EchartsProps } from '../types';
+import { getChartPadding, getLegendProps } from '../utils/series';
+
+/* eslint-disable no-param-reassign */
+function setLabelVisibility(nodes: GraphNodeItemOption[], showSymbolThreshold: number): void {
+  if (showSymbolThreshold > 0) {
+    nodes.forEach(function (node) {
+      node.label = {
+        show: node.value! > showSymbolThreshold,
+      };
+    });
+  }
+}
+
+/* eslint-disable no-param-reassign */
+function setNormalizedSymbolSize(nodes: GraphNodeItemOption[]): void {
+  let minValue: any = Number.MAX_VALUE;
+  let maxValue: any = Number.MIN_VALUE;
+  nodes.forEach(node => {
+    if (node.value! > maxValue) {
+      maxValue = node.value;
+    }
+    if (node.value! < minValue) {
+      minValue = node.value;
+    }
+  });
+  nodes.forEach(node => {
```

Comment:
Can we combine these two loops in one?

Reply:
I am not sure,how this can be done,as it need min and max of all nodes.

Reply:
or maybe this loop can be avoided altogether by storing values in seperate 1d array and using inbuild Match.min, max ?

Reply:
Right, didn't notice min/max is used in the second loop. I'm wondering whether you should just use `d3-array` and `d3-scale` here are they are designed for this type of data transformation.

This will add two new dependencies to the ECharts plugin, but maybe it's OK.

Reply:
This change is optional, too, and can be done in another PR if you agree it makes sense.

Reply:
I think calculating min,max can be removed by storing values in array and just using Math functions.

Reply:
Removed one loop,is it ok this way @ktmud ?

Reply:
Note `Math.min` and `Math.max` don't handle `null` & `undefined` well. 

I think it's OK to just use `d3-array`. We've used it in [many places](https://github.com/apache-superset/superset-ui/blob/57643c941ceee6daa700cb7a6c93fefbc8a845f9/plugins/plugin-chart-table/src/TableChart.tsx#L21) and Webpack should be smart enough to dedupe it during code-splitting.

Reply:
I was just bit hesitant to use extra dependancy just for min,max but I didn't know it can't handle falsey values.
Added d3-array.